src/button/style.ts

```diff
@@ -23,55 +23,80 @@ export const BUTTON_CSS = `
     display: none;
   }
   .kuc-button__button {
-    font-size: 16px;
+    font-size: var(--kuc-button-font-size, 16px);
+    width: var(--kuc-button-width, "auto");
+    height: var(--kuc-button-height, 48px);
     min-width: 163px;
-    height: 48px;
     padding: 0px 16px;
     user-select: none;
   }
   .kuc-button__button:focus {
     outline: none;
   }
   .kuc-button__button--normal {
+    background-color: var(--kuc-button-background-color, #f7f9fa);
+    color: var(--kuc-button-text-color, #3498db);
     border: 1px solid #e3e7e8;
-    background-color: #f7f9fa;
     box-shadow: 1px 1px 1px #ffffff inset;
-    color: #3498db;
   }
   .kuc-button__button--normal:hover,
   .kuc-button__button--normal:focus,
   .kuc-button__button--normal:active {
-    background-color: #c8d6dd;
     box-shadow: none;
     cursor: pointer;
   }
+  .kuc-button__button--normal:hover {
+    background-color: var(--kuc-button-background-hover, #c8d6dd);
+  }
+  .kuc-button__button--normal:focus {
+    background-color: var(--kuc-button-background-focus, #c8d6dd);
+  }
+  .kuc-button__button--normal:active {
+    background-color: var(--kuc-button-background-active, #c8d6dd);
+  }
   .kuc-button__button--submit {
+    background-color: var(--kuc-button-background-color, #3498db);
+    color: var(--kuc-button-text-color, #ffffff);
     border: 1px solid #e3e7e8;
-    background-color: #3498db;
-    color: #ffffff;
   }
   .kuc-button__button--submit:hover,
   .kuc-button__button--submit:focus,
   .kuc-button__button--submit:active {
-    background-color: #1d6fa5;
     cursor: pointer;
   }
+  .kuc-button__button--submit:hover {
+    background-color: var(--kuc-button-background-hover, #1d6fa5);
+  }
+  .kuc-button__button--submit:focus {
+    background-color: var(--kuc-button-background-focus, #1d6fa5);
+  }
+  .kuc-button__button--submit:active {
+    background-color: var(--kuc-button-background-active, #1d6fa5);
+  }
   .kuc-button__button--alert {
+    background-color: var(--kuc-button-background-color, #e74c3c);
+    color: var(--kuc-button-text-color, #ffffff);
     border: 0 none;
-    background-color: #e74c3c;
     box-shadow: 1px 1px 1px #ffffff inset;
-    color: #ffffff;
   }
   .kuc-button__button--alert:hover,
   .kuc-button__button--alert:focus,
   .kuc-button__button--alert:active {
-    background-color: #bf2718;
     box-shadow: none;
     cursor: pointer;
   }
+  .kuc-button__button--alert:hover {
+    background-color: var(--kuc-button-background-hover, #bf2718);
+  }
+  .kuc-button__button--alert:focus {
+    background-color: var(--kuc-button-background-focus, #bf2718);
+  }
+  .kuc-button__button--alert:active {
+    background-color: var(--kuc-button-background-active, #bf2718);
+  }
   .kuc-button__button:disabled {
+    background-color: var(--kuc-button-background-color, #d4d7d7);
     border: 1px solid #e3e7e8;
-    background-color: #d4d7d7;
     box-shadow: none;
     color: #888888;
```

Comment:
Do we need to update
color: var(--kuc-button-text-color, #888888);

Reply:
Yes, I think we should update it too